source/draw/README.md

```diff
@@ -39,3 +39,38 @@ Transforms go from zero in the center and increase with either pixels or pixels
 * **Reference transforms** start from the center of the image and increase in pixels. X goes right and Y goes down. This allows applying rotations with automatic compensation for aspect ratio and preserving pixel density when cropping images.
 
 * **Normalized transforms** start from the center of the image and go from (-1, -1) in the top left corner to (+1, +1) in the bottom right corner. This allows using the same coordinates for multiple image resolutions. The disadvantage is that it does not preserve aspect ratio when rotating since it treats all images as squares when not knowing their size. This is the normalized version of reference coordinates.
+
+# DrawState
+Using `DrawState` you can draw things without having to call the canvas directly.
+This is convenient if you have a lot of things to draw that have settings in common.
+
+## How to use
+Create a `DrawState` with a target image as input argument.
+Continue with setters to add information.
+The setters leaves no side-effects but returns a clone of the state with the setting changed.
+```ooc
+DrawState new(target) setMap(shader) draw()
```

Comment:
Later it might support raster images but only in CPU contexts then. Settings might not be fully shared.